cmd/tailscale/cli/network-lock.go

```diff
@@ -167,24 +183,65 @@ func runNetworkLockModify(ctx context.Context, addArgs, removeArgs []string) err
 
 var nlSignCmd = &ffcli.Command{
 	Name:       "sign",
-	ShortUsage: "sign <node-key>",
+	ShortUsage: "sign <node-key> [<rotation-key>]",
 	ShortHelp:  "Signs a node-key and transmits that signature to the control plane",
 	Exec:       runNetworkLockSign,
 }
 
-// TODO(tom): Implement specifying the rotation key for the signature.
 func runNetworkLockSign(ctx context.Context, args []string) error {
-	switch len(args) {
-	case 0:
-		return errors.New("expected node-key as second argument")
-	case 1:
-		var nodeKey key.NodePublic
-		if err := nodeKey.UnmarshalText([]byte(args[0])); err != nil {
-			return fmt.Errorf("decoding node-key: %w", err)
+	var (
+		nodeKey     key.NodePublic
+		rotationKey key.NLPublic
+	)
+
+	if len(args) == 0 {
+		return errors.New("usage: lock sign <node-key> [<rotation-key>]")
+	}
+	if err := nodeKey.UnmarshalText([]byte(args[0])); err != nil {
+		return fmt.Errorf("decoding node-key: %w", err)
+	}
+	if len(args) > 1 {
+		if err := rotationKey.UnmarshalText([]byte(args[1])); err != nil {
+			return fmt.Errorf("decoding rotation-key: %w", err)
 		}
+	}
+
+	return localClient.NetworkLockSign(ctx, nodeKey, []byte(rotationKey.Verifier()))
+}
+
+var nlDisableCmd = &ffcli.Command{
+	Name:       "disable",
+	ShortUsage: "disable <disablement-secret>",
+	ShortHelp:  "Consumes a disablement secret to shut down network-lock across the tailnet",
+	Exec:       runNetworkLockDisable,
+}
 
-		return localClient.NetworkLockSign(ctx, nodeKey, nil)
-	default:
-		return errors.New("expected a single node-key as only argument")
+func runNetworkLockDisable(ctx context.Context, args []string) error {
+	_, secrets, err := parseNLArgs(args, true)
+	if err != nil {
+		return err
+	}
+	if len(secrets) != 1 {
+		return errors.New("usage: lock disable <disablement-secret>")
+	}
+	return localClient.NetworkLockDisable(ctx, secrets[0])
+}
+
+var nlDisablementKDFCmd = &ffcli.Command{
+	Name:       "disablement-kdf",
+	ShortUsage: "disablement-kdf <hex-encoded-disablement-secret>",
```

Comment:
What is the value of doing this vs just using the hex string as the disablement  value?

Reply:
In the real flow for initialization, the UI will do the KDF computation for you and this wont be necessary. This is useful for now, before we have a UI.